pkg/ipcache/types.go

```diff
@@ -95,15 +114,24 @@ func (s prefixInfo) isValid() bool {
 	return false
 }
 
-func (s prefixInfo) ToLabels() labels.Labels {
+func (s PrefixInfo) sortedByResourceIDsAndSource() []types.ResourceID {
+	resourceIDs := maps.Keys(s)
+	slices.Sort(resourceIDs)
+	slices.SortFunc(resourceIDs, func(a, b types.ResourceID) bool {
+		return !source.AllowOverwrite(s[a].source, s[b].source)
+	})
```

Comment:
Hm, upon second thought, maybe we want to reverse this so that we get the strongest sources first to weakest source, but if any entries have matching sources, then those are sorted by resource ID (alphabetically).